src/pre-engagement-form/form-components/form-component.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+/**
+ * This is a helper wrapper component.
+ * It handles things that are common to all form-components:
+ * - React Hook Form Binding
+ * - Validation and Error Messaging
+ * - Component Label
+ */
+
+/* eslint-disable react/require-default-props */
+import React, { useRef } from 'react';
+import { connect } from 'react-redux';
+import { useFormContext, Controller, UseControllerProps, FieldError } from 'react-hook-form';
+
+import { useLocalization } from '../localization';
+import { setValue } from '../state';
+import { Label } from './styles';
+
+type OwnProps = {
+  label: string;
+  defaultValue?: string;
+  children: JSX.Element;
+};
+
+type Props = OwnProps & UseControllerProps & typeof mapDispatchToProps;
+
+// TODO: Displaying error can be more elegant and robust
+const getErrorMessage = (error: any | FieldError | undefined, strings: Record<string, string>) => {
+  if (error === undefined) {
+    return '';
+  }
+
+  if (error.type === 'required') {
+    return error.message || strings.FieldValidationRequiredField;
+  }
+
+  if (error.type === 'pattern') {
+    /**
+     * TODO: ideally there should be a a more generic error string like
+     * "FieldValidationInvalidPattern". Right now, Email Validation is the only
+     * validation we're using besides Required Fields, so we're fine with
+     * falling back to "FieldValidationInvalidEmail".
+     */
+    return error.message || strings.FieldValidationInvalidEmail;
+  }
+
+  return '';
```

Comment:
Maybe just return `error.message` in case there's an actual error?

Reply:
that would be a good fallback, I'll update the last line to be `error.message || ''`